crates/shared/src/event_handling.rs

```diff
@@ -204,6 +204,28 @@ where
             });
         }
 
+        // Special case where multiple new blocks were added and no reorg happened.
+        // Because we need to fetch the full block range we only do this if the number
+        // of new blocks is sufficiently small.
+        let block_range = RangeInclusive::try_new(last_handled_block_number, current_block_number)?;
```

Comment:
I think this will return error with `?` in cases when reorg happens such that the new chain is shorter than the old chain (block height becomes lower which means `current_block_number` < `last_handled_block_number`).

This obviously depends on the fork choice rule and we would need to be sure that every chain we deploy to doesn't allow for this to happen.

So, to be safe, I would adjust this to:

```suggestion
        if let Ok(block_range) = RangeInclusive::try_new(last_handled_block_number, current_block_number)
        {
            // rest
        }
```

Reply:
I though the fork choice rule would always prefer longer chains.
In any case since this code is magic to me your extra safe guard makes sense to me. 👍